torchvision/prototype/datasets/_builtin/sintel.py

```diff
@@ -0,0 +1,180 @@
+
+    def _filter_split(self, data: Tuple[str, Any], *, split: str) -> bool:
+        path = pathlib.Path(data[0])
+        # The dataset contains has the folder "training", while allowed options for `split` are
+        # "train" and "test", we don't check for equality here ("train" != "training") and instead
+        # check if split is in the folder name
+        return split in path.parents[2].name
+
+    def _filter_images(self, data: Tuple[str, Any], *, pass_name: str) -> bool:
+        path = pathlib.Path(data[0])
+        regexp = self._PASS_NAME_BOTH if pass_name == "both" else re.compile(pass_name)
+        matched = bool(regexp.search(str(path.parent)))
+        return matched and path.suffix == ".png"
+
+    def _classify_archive(self, data: Tuple[str, Any], *, pass_name: str) -> Optional[int]:
+        path = pathlib.Path(data[0])
+        suffix = path.suffix
+        if suffix == ".flo":
+            return 0
+        elif suffix == ".png":
+            regexp = self._PASS_NAME_BOTH if pass_name == "both" else re.compile(pass_name)
+            matched = regexp.search(path.parents[1].name)
+            return 1 if matched else None
+        else:
+            return None
+
+    def _read_flo(self, file: io.IOBase) -> torch.Tensor:
+        magic = file.read(4)
+        if magic != b"PIEH":
+            raise ValueError("Magic number incorrect. Invalid .flo file")
+        w = int.from_bytes(file.read(4), "little")
+        h = int.from_bytes(file.read(4), "little")
+        data = file.read(2 * w * h * 4)
+        data_arr = np.frombuffer(data, dtype=np.float32)
+        # Creating a copy of the underlying array, to avoid UserWarning: "The given NumPy array
+        #     is not writeable, and PyTorch does not support non-writeable tensors."
+        return torch.from_numpy(np.copy(data_arr.reshape(h, w, 2).transpose(2, 0, 1)))
+
+    def _flows_key(self, data: Tuple[str, Any]) -> Tuple[str, int]:
+        path = pathlib.Path(data[0])
+        category = path.parent.name
+        idx = int(self._FILE_NAME_PATTERN.match(path.name).group("idx"))  # type: ignore[union-attr]
+        return category, idx
+
+    def _images_key(self, data: Tuple[Tuple[str, Any], Tuple[str, Any]]) -> Tuple[str, int]:
+        return self._flows_key(data[0])
+
+    def _collate_and_decode_sample(
+        self,
+        data: Tuple[Tuple[str, io.IOBase], Any],
+        *,
+        decoder: Optional[Callable[[io.IOBase], torch.Tensor]],
+        config: DatasetConfig,
+    ) -> Dict[str, Any]:
+        flo, images = data
+        img1, img2 = images
+        flow_arr = self._read_flo(flo[1]) if flo[1] else None
+
+        path1, buffer1 = img1
+        path2, buffer2 = img2
+
+        return dict(
+            image1=decoder(buffer1) if decoder else buffer1,
+            image1_path=path1,
+            image2=decoder(buffer2) if decoder else buffer2,
+            image2_path=path2,
+            flow=flow_arr,
+            flow_path=flo[0] or "",
```

Comment:
Returning `None` is fine

```suggestion
            flow_path=flo[0],
```